### PR TITLE
Pin numba to < 0.59 due to pending @generated_jit deprecation

### LINKED ISSUE #280: Pin release version of numba to < 0.59.0
Codex version: 0.3.4

Raised downstream by crystalball. I see generated_jit is being removed in the near future so I'm marking this and make a patch release of codex to ensure the downstream packages don't start failing -- I don't think there is time now to modify the codebase to use @overload. Per docs we should pin to to below that version.

### PR DESCRIPTION
**Pull request by sjperkins**
Closes #280 

Supercedes:
 - #281 

Related to:
 - #283 

- [x] Tests added / passed

  ```bash
  $ py.test -v -s africanus
  ```

  If the pep8 tests fail, the quickest way to correct
  this is to run `autopep8` and then `flake8` and
  `pycodestyle` to fix the remaining issues.

  ```
  $ pip install -U autopep8 flake8 pycodestyle
  $ autopep8 -r -i africanus
  $ flake8 africanus
  $ pycodestyle africanus
  ```

- [x] Fully documented, including `HISTORY.rst` for all changes
      and one of the `docs/*-api.rst` files for new API

  To build the docs locally:

  ```
  pip install -r requirements.readthedocs.txt
  cd docs
  READTHEDOCS=True make html
  ```

### CODE REVIEW

**Comment by sjperkins** [comment]
@bennahugo This is not significantly different from #281, so I'll merge at the end of the day if you don't have time to review by then.

**Review by bennahugo — APPROVED** [review]

**Comment by bennahugo** [comment]
Thanks @sjperkins. Yup I put the environment variable in the wrong place by the looks of it.

**Pull request merged**
